geekodoc/tests/run-tests.sh

```diff
@@ -17,55 +17,52 @@ PROGDIR=${0%/*}
 SCHEMA=${PROGDIR}/../rng/geekodoc5-flat.rng
 SCHEMA=$(readlink -f ${SCHEMA})
 ERRORS=0
+ERRORFILE=${PROGDIR}/last-test-run-errors
 
 SUCCESS=0
 FAILURE=1
 
+REMAKE=1
+
 TEST_CATEGORY=all
 
 export LANG=C
 
 function logerror() {
+    local fatal=0
+    if [[ $1 == '--fatal' ]]; then
+        fatal=1
+        shift
+    fi
     echo -e "\e[1;31mERROR: $1\e[0m"
+    [[ $fatal -ne 0 ]] && exit 1
 }
 
 function loginfo() {
-    echo -e "\e[1;39mINFO: $1\e[0m"
+    local extraoptions=
+    if [[ $1 == '-n' ]]; then
+        extraoptions='-n'
+        shift
+    fi
+    echo -e $extraoptions "\e[1;39mINFO: $1\e[0m"
 }
 
 function logwarn() {
     echo -e "\e[1;95mWARN: $1\e[0m"
 }
 
-function validate_with_jing {
-    local _RNG=$1
-    local _XML=$2
-    local _ERROR=${2%*.xml}.err
-    local result
-    jing $_RNG $_XML >$_ERROR
-    return $?
-}
-
-function validate_with_xmllint {
-    local _RNG=$1
-    local _XML=$2
-    local _ERROR=${2%*.xml}.err
-    xmllint --noout --relaxng $_RNG $_XML 2>$_ERROR
-    return $?
-}
-
-function validator {
+function validator() {
     case "$VALIDATOR" in
```

Comment:
Yes, that's probably better to have only one function for validation. The only thing that I would like to change is returning the return code (with `return $?`) as I've mentioned in my previous comment.

Reply:
Yeah, I don't think you can get both error message as a variable and the return code too, unfortunately.